src/modules/esl-popup/core/esl-popup.ts

```diff
@@ -74,88 +96,121 @@ export class ESLPopup extends ESLToggleable {
+    if (entry.intersectionRect.y !== entry.boundingClientRect.y) {
+      this._intersectionRatio.top = entry.intersectionRect.height / entry.boundingClientRect.height;
+      this._checkIntersectionForAdjacentAxis(this._isPositioningAlongHorizontal, this._intersectionRatio.top);
+    }
+    if (entry.intersectionRect.bottom !== entry.boundingClientRect.bottom) {
+      this._intersectionRatio.bottom = entry.intersectionRect.height / entry.boundingClientRect.height;
+      this._checkIntersectionForAdjacentAxis(this._isPositioningAlongHorizontal, this._intersectionRatio.bottom);
+    }
+    if (entry.intersectionRect.x !== entry.boundingClientRect.x) {
+      this._intersectionRatio.left = entry.intersectionRect.width / entry.boundingClientRect.width;
+      this._checkIntersectionForAdjacentAxis(this._isPositioningAlongVertical, this._intersectionRatio.left);
+    }
+    if (entry.intersectionRect.right !== entry.boundingClientRect.right) {
+      this._intersectionRatio.right = entry.intersectionRect.width / entry.boundingClientRect.width;
+      this._checkIntersectionForAdjacentAxis(this._isPositioningAlongVertical, this._intersectionRatio.right);
     }
   }
 
-  protected _calculatePosition($activator: HTMLElement) {
-    const {left, arrowLeft} = this._calculateLeft($activator);
-    const {top, arrowTop, position} = this._calculateTop($activator);
-
-    return {
-      left,
-      top,
-      arrowLeft,
-      arrowTop,
-      position
-    };
+  @bind
+  protected onActivatorScroll(e: Event) {
+    this._updatePosition();
   }
 
-  protected _calculateLeft($activator: HTMLElement) {
-    const triggerRect = $activator.getBoundingClientRect();
-    const triggerPosX = triggerRect.left + window.pageXOffset;
-    const centerX = triggerPosX + triggerRect.width / 2;
+  protected _addActivatorObserver(target: HTMLElement) {
+    const scrollParents = getListScrollParents(target);
 
-    let arrowAdjust = 0;
-    let left = centerX - this.offsetWidth / 2;
+    const unsubscribers = scrollParents.map(($root) => {
+      const options = {passive: true} as EventListenerOptions;
+      $root.addEventListener('scroll', this.onActivatorScroll, options);
+      return () => {
+        $root && $root.removeEventListener('scroll', this.onActivatorScroll, options);
+      };
+    });
 
-    if (this.behavior === 'fit' && left < this._offsetWindow) {
-      arrowAdjust += left - this._offsetWindow;
-      left = this._offsetWindow;
-    }
+    const options = {
+      rootMargin: '0px',
+      threshold: [...Array(9).keys()].map((x) => x/8)
+    } as IntersectionObserverInit;
 
-    const right = this._windowWidth - (left + this.offsetWidth);
-    if (this.behavior === 'fit' && right < this._offsetWindow) {
-      arrowAdjust -= right - this._offsetWindow;
-      left += right - this._offsetWindow;
-    }
-    const arrowLeft = this.clientWidth / 2 + arrowAdjust;
+    const observer = new IntersectionObserver(this.onActivatorIntersection, options);
```

Comment:
It will make sense to move IObserver to a statically defined (not compulsory the class static member, but at least make it single across popup instances)
At least, in case we are not going to declare options dependently of instance
IObserver prefers to be shared across consumers instead of being created separately for each instance. It will simplify unsubscription by the way...

Reply:
Also, not everything is clear here.
For example, as a future feature... Is it possible for popups to set different rootMargin? It was just that this item was raised during the discussions.
Let's not rush for now and will discuss it?

Reply:
Yep, of cose. Thats just a point to discuss in future.